src/ast/mod.rs

```diff
@@ -857,28 +857,32 @@ impl fmt::Display for ListAgg {
 
 /// The `ON OVERFLOW` clause of a LISTAGG invocation
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct ListAggOnOverflow {
-    pub error: bool,
-    pub filler: Option<Box<Expr>>,
-    pub with_count: bool,
+pub enum ListAggOnOverflow {
+    Error,
+    Truncate {
+        filler: Option<Box<Expr>>,
+        with_count: bool,
+    },
 }
 
 impl fmt::Display for ListAggOnOverflow {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let overflow = if self.error {
-            "ERROR".to_owned()
-        } else {
-            format!(
-                "TRUNCATE {}{} COUNT",
-                if let Some(filler) = &self.filler {
-                    format!("{} ", filler)
+        write!(f, " ON OVERFLOW")?;
+        match self {
+            ListAggOnOverflow::Error => write!(f, " ERROR"),
+            ListAggOnOverflow::Truncate { filler, with_count } => {
+                write!(f, " TRUNCATE")?;
+                if let Some(filler) = filler {
+                    write!(f, " {}", filler)?;
+                }
+                if *with_count {
```

Comment:
Perhaps I value vertical space too much, but I'd write this as: `write!(f, " {} COUNT", if *with_count { "WITH" } else { "WITHOUT" })`. Leaving it as is is fine too.

Reply:
I left it for now, but happy to clean it up later.